### DISCUSSION TITLE
Sorting in igx-column

### DISCUSSION THREAD

**Discussion by kevalgw** [IgniteUI/igniteui-angular#11888] [Q&A] [answered]
Hey
I want to sort the igx-column but the data of the columnn is in object type
eg: tableData {
someData: { name: string, model: string, name: string};
status: string;
registration: string;
action: string;
}
Considering above data in column someData i want to sort it by key "name" ..
How Do i Do it??
<igx-column field="someData" [sortable]="true">
the above code is not working

**Comment by RivaIvanova** [comment, marked as answer]
Hello @kevalgw,

An approach I could suggest is implementing your own **custom sorting strategy**. Sorting is performed using our [DefaultSortingStrategy](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/defaultsortingstrategy.html) algorithm, however, any **IgxColumnComponent** or **ISortingExpression** can use a custom implementation of the [ISortingStrategy](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/interfaces/isortingstrategy.html) as a substitute algorithm.

Additionally, I have prepared a small sample demonstrating the above approach and it could be found [here](https://stackblitz.com/edit/angular-en8rch-qtpkuh?file=src%2Fapp%2Fsample%2Figx-sample.component.ts).

**Reply by kevalgw** [reply]
Hi @RivaIvanova 

Thanks for the reply , I appreciate.
I am getting the following error in my code.
 
**error TS1005: ';' expected.

411   public override compareValues(obj1: any, obj2: any): any {**

![image](https://user-images.githubusercontent.com/101707048/181447740-25ad3e0c-6524-41f7-a19d-f149a56d613e.png)

**Reply by RivaIvanova** [reply]
Could you provide a sample that demonstrates the behavior on your side? Also, it would be great if you could try and modify the provided by me sample.

**Reply by kevalgw** [reply]
Hi @RivaIvanova 

I have attached the ss of the function written outside of main class.

![image](https://user-images.githubusercontent.com/101707048/181690378-a3b55698-7183-4371-8423-2b39358b390e.png)


And Imported it in main class

![image](https://user-images.githubusercontent.com/101707048/181690548-01304fcd-e59d-47d9-884e-fd6b6bbce170.png)


Still it is giving me error

**Reply by RivaIvanova** [reply]
Hello @kevalgw,

I can see that the custom sorting strategy is implemented similarly to the suggested one, however, as the isolated sample from my previous message seems to be working correctly and cannot reproduce the issue, it is hard to tell what the cause for this behavior could be without having a working sample on my side which I can debug. 

It would be very helpful if you could fork the previous [sample](https://stackblitz.com/edit/angular-en8rch-qtpkuh?file=src%2Fapp%2Fsample%2Figx-sample.component.ts), modify it so it reproduces the issue, and send it back.

**Reply by kevalgw** [reply]
Hi @RivaIvanova 

I tried the same example with new project, its working properly.
But in my existing project it is not working, I guess its version issue 

Following are the versions installed 

"igniteui-angular": "^10.2.0"
"@angular/cdk": "^10.2.5",

Please check whether its the issue because of version or not

**Reply by RivaIvanova** [reply]
Hello @kevalgw,

From the provided information what I can say is that this is not related to our product, it is TypeScript-related and what I can suggest is omitting the **override** keyword. As mentioned in TypeScript's official documentation [here](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-3.html#:~:text=That%E2%80%99s%20why%20TypeScript%204.3%20adds%20the%20override%20keyword.), this keyword is introduced in **TypeScript 4.3.** and in case the version used on your side is lower, then this behavior is expected.

Additionally, please keep in mind that version **10.2.0 is considered retired**.

**Reply by kevalgw** [reply]
Hi @RivaIvanova 

It worked for me by removing override keyword. 
Thanks a lot and I appreciate for your support.
